specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx

```diff
@@ -133,12 +180,28 @@ export function PrepDialog({
           max={maxPrep}
           min={0}
           title={interactionsText.selectedAmount()}
-          value={bulkValue}
+          value={bulkValue || 0}
           onValueChange={(newCount): void => {
-            setBulkValue(newCount);
-            setSelected(
-              preparations.map(({ available }) => Math.min(available, newCount))
-            );
+            if (newCount === '') {
+              setBulkValue(0);
+              setSelected((prev) => {
+                const newSelected = { ...prev };
+                sortedPreparations.forEach((prep) => {
+                  newSelected[prep.preparationId] = null;
+                });
+                return newSelected;
+              });
+            } else {
+              const count = parseInt(newCount, 10);
```

Comment:
hmm, the `Input.Integer` is supposed to call `onValueChange` with `number` not `string` as far as I remember
this this parseInt here and the `if (newCount === '') {` check above is not right 🤔